builtin/v12/migration/miner.go

```diff
@@ -224,65 +271,112 @@ delta = prevInRoot - inRoot
 
 merge( prevOutRoot, Migrated(delta) )
 */
-func migrateSectorsWithDiff(ctx context.Context, store adt11.Store, inRoot cid.Cid, prevInRoot cid.Cid, prevOutRoot cid.Cid) (cid.Cid, error) {
-	// we have previous work, but the AMT has changed -- diff them
+
+// todo modify hamt based on diff
+func (m minerMigrator) migrateSectorsWithDiff(ctx context.Context, store adt11.Store, minerAddr address.Address, inRoot cid.Cid, prevInRoot cid.Cid, prevOutRoot cid.Cid, prevSectorIndexRoot cid.Cid) (cid.Cid, cid.Cid, error) {
```

Comment:
Not a requirement for merging but I suspect this is going to come out overall cleaner if you share code with the migrateDeadlineSectorsWithDiff function.  I would probably keep two separate function calls with the same interface you have and then call into a shared internal function which takes a boolean flag for operating over the sectorNumber => DealID index.

Reply:
Same thing applies to the separate migrateDeadlineSectorsFromScratch function

Reply:
it's a six vs half a dozen situation, going to leave it as is